src/Config.ts

```diff
@@ -3,62 +3,59 @@ import Debug from 'debug'
 import { ControlLayer, MessageLayer } from 'streamr-client-protocol'
 import { ExternalProvider, JsonRpcFetchFunc } from '@ethersproject/providers'
 import { BigNumber } from '@ethersproject/bignumber'
-import { O } from 'ts-toolbelt'
 import { getVersionString, counterId } from './utils'
 import { Todo } from './types'
 
 export type EthereumConfig = ExternalProvider|JsonRpcFetchFunc
 
 export type StreamrClientOptions = {
     id?: string
```

Comment:
Removed `id` and `debug` fields from options as those were duplicate to `StreamrClient`'s fields. Neither of the fields affected the `id` and `debug` fields of the `StreamrClient` instance, so in practice those were not configuration options.

Changed `Connection` and `WebSocket` to get the debug instance from `StreamrClient`. 

`AuthFetch` creates its own debug instance. If we later refactor it to a class, it can also get the debug instance from `StreamrClient`.